// @aws-sdk/client-location/dist-es/schemas/schemas_0.js
const _A = "Accuracy";
const _AA = "AndroidApp";
const _AAA = "AllowAndroidApps";
const _AAAl = "AllowAppleApps";
const _AAL = "AndroidAppList";
const _AALp = "AppleAppList";
const _AAl = "AllowActions";
const _AAp = "AppleApp";
const _ADE = "AccessDeniedException";
const _AF = "AvoidFerries";
const _AK = "ApiKey";
const _AKF = "ApiKeyFilter";
const _AKR = "ApiKeyRestrictions";
const _AN = "AddressNumber";
const _AR = "AllowResources";
const _ARl = "AllowReferers";
const _AT = "AvoidTolls";
const _ATC = "AssociateTrackerConsumer";
const _ATCR = "AssociateTrackerConsumerRequest";
const _ATCRs = "AssociateTrackerConsumerResponse";
const _ATr = "ArrivalTime";
const _B = "Blob";
const _BB = "BoundingBox";
const _BDDPH = "BatchDeleteDevicePositionHistory";
const _BDDPHE = "BatchDeleteDevicePositionHistoryError";
const _BDDPHEL = "BatchDeleteDevicePositionHistoryErrorList";
const _BDDPHR = "BatchDeleteDevicePositionHistoryRequest";
const _BDDPHRa = "BatchDeleteDevicePositionHistoryResponse";
const _BDG = "BatchDeleteGeofence";
const _BDGE = "BatchDeleteGeofenceError";
const _BDGEL = "BatchDeleteGeofenceErrorList";
const _BDGR = "BatchDeleteGeofenceRequest";
const _BDGRa = "BatchDeleteGeofenceResponse";
const _BEG = "Base64EncodedGeobuf";
const _BEGE = "BatchEvaluateGeofencesError";
const _BEGEL = "BatchEvaluateGeofencesErrorList";
const _BEGR = "BatchEvaluateGeofencesRequest";
const _BEGRa = "BatchEvaluateGeofencesResponse";
const _BEGa = "BatchEvaluateGeofences";
const _BGDP = "BatchGetDevicePosition";
const _BGDPE = "BatchGetDevicePositionError";
const _BGDPEL = "BatchGetDevicePositionErrorList";
const _BGDPR = "BatchGetDevicePositionRequest";
const _BGDPRa = "BatchGetDevicePositionResponse";
const _BI = "BundleId";
const _BIE = "BatchItemError";
const _BP = "BiasPosition";
const _BPG = "BatchPutGeofence";
const _BPGE = "BatchPutGeofenceError";
const _BPGEL = "BatchPutGeofenceErrorList";
const _BPGR = "BatchPutGeofenceRequest";
const _BPGRE = "BatchPutGeofenceRequestEntry";
const _BPGREL = "BatchPutGeofenceRequestEntryList";
const _BPGRa = "BatchPutGeofenceResponse";
const _BPGS = "BatchPutGeofenceSuccess";
const _BPGSL = "BatchPutGeofenceSuccessList";
const _BUDP = "BatchUpdateDevicePosition";
const _BUDPE = "BatchUpdateDevicePositionError";
const _BUDPEL = "BatchUpdateDevicePositionErrorList";
const _BUDPR = "BatchUpdateDevicePositionRequest";
const _BUDPRa = "BatchUpdateDevicePositionResponse";
const _C = "Code";
const _CA = "ConsumerArn";
const _CAa = "CalculatorArn";
const _CAo = "CollectionArn";
const _CAon = "ConsumerArns";
const _CC = "CountryCode3";
const _CCL = "CountryCodeList";
const _CCOE = "CountryCode3OrEmpty";
const _CC_ = "Cache-Control";
const _CCa = "CacheControl";
const _CE = "ConflictException";
const _CF = "CertificateFingerprint";
const _CGC = "CreateGeofenceCollection";
const _CGCR = "CreateGeofenceCollectionRequest";
const _CGCRr = "CreateGeofenceCollectionResponse";
const _CI = "CellId";
const _CK = "CreateKey";
const _CKR = "CreateKeyRequest";
const _CKRr = "CreateKeyResponse";
const _CL = "CustomLayers";
const _CM = "CreateMap";
const _CMO = "CarModeOptions";
const _CMR = "CreateMapRequest";
const _CMRr = "CreateMapResponse";
const _CN = "CollectionName";
const _CNa = "CalculatorName";
const _CPI = "CreatePlaceIndex";
const _CPIR = "CreatePlaceIndexRequest";
const _CPIRr = "CreatePlaceIndexResponse";
const _CR = "CalculateRoute";
const _CRC = "CreateRouteCalculator";
const _CRCMO = "CalculateRouteCarModeOptions";
const _CRCR = "CreateRouteCalculatorRequest";
const _CRCRr = "CreateRouteCalculatorResponse";
const _CRM = "CalculateRouteMatrix";
const _CRMR = "CalculateRouteMatrixRequest";
const _CRMRa = "CalculateRouteMatrixResponse";
const _CRMS = "CalculateRouteMatrixSummary";
const _CRR = "CalculateRouteRequest";
const _CRRa = "CalculateRouteResponse";
const _CRS = "CalculateRouteSummary";
const _CRTMO = "CalculateRouteTruckModeOptions";
const _CS = "CellSignals";
const _CT = "CreateTime";
const _CTR = "CreateTrackerRequest";
const _CTRr = "CreateTrackerResponse";
const _CT_ = "Content-Type";
const _CTo = "ContentType";
const _CTr = "CreateTracker";
const _CU = "ConfigurationUpdate";
const _Ca = "Categories";
const _Ce = "Center";
const _Ci = "Circle";
const _Co = "Configuration";
const _Cou = "Country";
const _D = "Distance";
const _DD = "DeviationDistance";
const _DGC = "DeleteGeofenceCollection";
const _DGCR = "DeleteGeofenceCollectionRequest";
const _DGCRe = "DeleteGeofenceCollectionResponse";
const _DGCRes = "DescribeGeofenceCollectionRequest";
const _DGCResc = "DescribeGeofenceCollectionResponse";
const _DGCe = "DescribeGeofenceCollection";
const _DI = "DeviceId";
const _DIe = "DeviceIds";
const _DK = "DeleteKey";
const _DKR = "DeleteKeyRequest";
const _DKRe = "DeleteKeyResponse";
const _DKRes = "DescribeKeyRequest";
const _DKResc = "DescribeKeyResponse";
const _DKe = "DescribeKey";
const _DM = "DeleteMap";
const _DMR = "DeleteMapRequest";
const _DMRe = "DeleteMapResponse";
const _DMRes = "DescribeMapRequest";
const _DMResc = "DescribeMapResponse";
const _DMe = "DescribeMap";
const _DN = "DepartNow";
const _DP = "DevicePositions";
const _DPI = "DeletePlaceIndex";
const _DPIR = "DeletePlaceIndexRequest";
const _DPIRe = "DeletePlaceIndexResponse";
const _DPIRes = "DescribePlaceIndexRequest";
const _DPIResc = "DescribePlaceIndexResponse";
const _DPIe = "DescribePlaceIndex";
const _DPL = "DevicePositionList";
const _DPU = "DevicePositionUpdates";
const _DPUL = "DevicePositionUpdateList";
const _DPUe = "DevicePositionUpdate";
const _DPe = "DeparturePositions";
const _DPep = "DeparturePosition";
const _DPes = "DestinationPositions";
const _DPest = "DestinationPosition";
const _DPev = "DevicePosition";
const _DRC = "DeleteRouteCalculator";
const _DRCR = "DeleteRouteCalculatorRequest";
const _DRCRe = "DeleteRouteCalculatorResponse";
const _DRCRes = "DescribeRouteCalculatorRequest";
const _DRCResc = "DescribeRouteCalculatorResponse";
const _DRCe = "DescribeRouteCalculator";
const _DS = "DataSource";
const _DSC = "DataSourceConfiguration";
const _DSe = "DeviceState";
const _DSu = "DurationSeconds";
const _DT = "DepartureTime";
const _DTC = "DisassociateTrackerConsumer";
const _DTCR = "DisassociateTrackerConsumerRequest";
const _DTCRi = "DisassociateTrackerConsumerResponse";
const _DTR = "DeleteTrackerRequest";
const _DTRe = "DeleteTrackerResponse";
const _DTRes = "DescribeTrackerRequest";
const _DTResc = "DescribeTrackerResponse";
const _DTe = "DeleteTracker";
const _DTes = "DescribeTracker";
const _DU = "DistanceUnit";
const _De = "Description";
const _Di = "Dimensions";
const _E = "Error";
const _EBE = "EventBridgeEnabled";
const _EC = "ErrorCount";
const _EI = "EventId";
const _EP = "EndPosition";
const _ET = "ExpireTime";
const _ETE = "EndTimeExclusive";
const _ETv = "EventType";
const _Ea = "Earfcn";
const _En = "Entries";
const _Er = "Errors";
const _F = "Filter";
const _FBB = "FilterBBox";
const _FBT = "ForecastedBreachTime";
const _FC = "FilterCountries";
const _FCi = "FilterCategories";
const _FD = "ForceDelete";
const _FE = "ForecastedEvent";
const _FEL = "ForecastedEventsList";
const _FEo = "ForecastedEvents";
const _FG = "FilterGeometry";
const _FGE = "ForecastGeofenceEvents";
const _FGEDS = "ForecastGeofenceEventsDeviceState";
const _FGER = "ForecastGeofenceEventsRequest";
const _FGERo = "ForecastGeofenceEventsResponse";
const _FL = "FieldList";
const _FN = "FileName";
const _FPCL = "FilterPlaceCategoryList";
const _FS = "FontStack";
const _FU = "ForceUpdate";
const _FUR = "FontUnicodeRange";
const _G = "Geometry";
const _GC = "GeofenceCount";
const _GDP = "GetDevicePosition";
const _GDPH = "GetDevicePositionHistory";
const _GDPHR = "GetDevicePositionHistoryRequest";
const _GDPHRe = "GetDevicePositionHistoryResponse";
const _GDPR = "GetDevicePositionRequest";
const _GDPRe = "GetDevicePositionResponse";
const _GG = "GeofenceGeometry";
const _GGR = "GetGeofenceRequest";
const _GGRe = "GetGeofenceResponse";
const _GGe = "GetGeofence";
const _GI = "GeofenceId";
const _GIe = "GeofenceIds";
const _GMG = "GetMapGlyphs";
const _GMGR = "GetMapGlyphsRequest";
const _GMGRe = "GetMapGlyphsResponse";
const _GMS = "GetMapSprites";
const _GMSD = "GetMapStyleDescriptor";
const _GMSDR = "GetMapStyleDescriptorRequest";
const _GMSDRe = "GetMapStyleDescriptorResponse";
const _GMSR = "GetMapSpritesRequest";
const _GMSRe = "GetMapSpritesResponse";
const _GMT = "GetMapTile";
const _GMTR = "GetMapTileRequest";
const _GMTRe = "GetMapTileResponse";
const _GO = "GeometryOffset";
const _GP = "GeofenceProperties";
const _GPR = "GetPlaceRequest";
const _GPRe = "GetPlaceResponse";
const _GPe = "GetPlace";
const _Ge = "Geobuf";
const _H = "Horizontal";
const _He = "Height";
const _I = "Interpolated";
const _IA = "IndexArn";
const _IAp = "Ipv4Address";
const _IDIG = "IsDeviceInGeofence";
const _ILG = "IncludeLegGeometry";
const _IN = "IndexName";
const _IS = "InferredState";
const _ISE = "InternalServerException";
const _IU = "IntendedUse";
const _K = "Key";
const _KA = "KeyArn";
const _KKEGQ = "KmsKeyEnableGeospatialQueries";
const _KKI = "KmsKeyId";
const _KN = "KeyName";
const _KS = "KeyStatus";
const _L = "Legs";
const _LCD = "LteCellDetails";
const _LCDL = "LteCellDetailsList";
const _LDP = "ListDevicePositions";
const _LDPR = "ListDevicePositionsRequest";
const _LDPRE = "ListDevicePositionsResponseEntry";
const _LDPREL = "ListDevicePositionsResponseEntryList";
const _LDPRi = "ListDevicePositionsResponse";
const _LG = "LegGeometry";
const _LGC = "ListGeofenceCollections";
const _LGCR = "ListGeofenceCollectionsRequest";
const _LGCRE = "ListGeofenceCollectionsResponseEntry";
const _LGCREL = "ListGeofenceCollectionsResponseEntryList";
const _LGCRi = "ListGeofenceCollectionsResponse";
const _LGR = "ListGeofencesRequest";
const _LGRE = "ListGeofenceResponseEntry";
const _LGREL = "ListGeofenceResponseEntryList";
const _LGRi = "ListGeofencesResponse";
const _LGi = "ListGeofences";
const _LI = "LocalId";
const _LK = "ListKeys";
const _LKR = "ListKeysRequest";
const _LKRE = "ListKeysResponseEntry";
const _LKREL = "ListKeysResponseEntryList";
const _LKRi = "ListKeysResponse";
const _LL = "LegList";
const _LLI = "LteLocalId";
const _LM = "ListMaps";
const _LMR = "ListMapsRequest";
const _LMRE = "ListMapsResponseEntry";
const _LMREL = "ListMapsResponseEntryList";
const _LMRi = "ListMapsResponse";
const _LNM = "LteNetworkMeasurements";
const _LNML = "LteNetworkMeasurementsList";
const _LPI = "ListPlaceIndexes";
const _LPIR = "ListPlaceIndexesRequest";
const _LPIRE = "ListPlaceIndexesResponseEntry";
const _LPIREL = "ListPlaceIndexesResponseEntryList";
const _LPIRi = "ListPlaceIndexesResponse";
const _LR = "LinearRing";
const _LRC = "ListRouteCalculators";
const _LRCR = "ListRouteCalculatorsRequest";
const _LRCRE = "ListRouteCalculatorsResponseEntry";
const _LRCREL = "ListRouteCalculatorsResponseEntryList";
const _LRCRi = "ListRouteCalculatorsResponse";
const _LRi = "LinearRings";
const _LS = "LineString";
const _LT = "ListTrackers";
const _LTC = "ListTrackerConsumers";
const _LTCR = "ListTrackerConsumersRequest";
const _LTCRi = "ListTrackerConsumersResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTR = "ListTrackersRequest";
const _LTRE = "ListTrackersResponseEntry";
const _LTREL = "ListTrackersResponseEntryList";
const _LTRi = "ListTrackersResponse";
const _La = "Language";
const _Lab = "Label";
const _Le = "Leg";
const _Len = "Length";
const _M = "Message";
const _MA = "MapArn";
const _MAa = "MacAddress";
const _MC = "MapConfiguration";
const _MCU = "MapConfigurationUpdate";
const _MLR = "MultiLinearRings";
const _MN = "MapName";
const _MP = "MultiPolygon";
const _MR = "MaxResults";
const _Mc = "Mcc";
const _Mn = "Mnc";
const _Mu = "Municipality";
const _N = "Neighborhood";
const _NC = "NrCapable";
const _ND = "NearestDistance";
const _NE = "NoExpiry";
const _NM = "NetworkMeasurements";
const _NT = "NextToken";
const _Na = "Name";
const _O = "Offset";
const _OF = "OptimizeFor";
const _P = "Package";
const _PA = "PositionalAccuracy";
const _PC = "PlaceCategory";
const _PCL = "PlaceCategoryList";
const _PCo = "PostalCode";
const _PD = "ProxyDetected";
const _PF = "PositionFiltering";
const _PG = "PlaceGeometry";
const _PGR = "PutGeofenceRequest";
const _PGRu = "PutGeofenceResponse";
const _PGu = "PutGeofence";
const _PI = "PlaceId";
const _PL = "PositionList";
const _PM = "PropertyMap";
const _PP = "PricingPlan";
const _PPDS = "PricingPlanDataSource";
const _PPM = "PositionPropertyMap";
const _PPo = "PositionProperties";
const _PSC = "PlaceSupplementalCategory";
const _PSCL = "PlaceSupplementalCategoryList";
const _PV = "PoliticalView";
const _Pc = "Pci";
const _Pl = "Place";
const _Po = "Position";
const _Poi = "Point";
const _Pol = "Polygon";
const _R = "Radius";
const _RA = "ResourceArn";
const _RBB = "RouteBBox";
const _RBBe = "ResultBBox";
const _RC = "RouteCount";
const _RM = "RouteMatrix";
const _RME = "RouteMatrixEntry";
const _RMEE = "RouteMatrixEntryError";
const _RMR = "RouteMatrixRow";
const _RNFE = "ResourceNotFoundException";
const _RP = "RefererPattern";
const _RPL = "RefererPatternList";
const _RT = "ReceivedTime";
const _Re = "Restrictions";
const _Rea = "Reason";
const _Reg = "Region";
const _Rel = "Relevance";
const _Res = "Results";
const _Rs = "Rsrp";
const _Rsr = "Rsrq";
const _Rss = "Rss";
const _S = "Successes";
const _SB = "SensitiveBoolean";
const _SC = "SupplementalCategories";
const _SD = "SensitiveDouble";
const _SDP = "SnappedDeparturePositions";
const _SDPn = "SnappedDestinationPositions";
const _SFPR = "SearchForPositionResult";
const _SFPRL = "SearchForPositionResultList";
const _SFSR = "SearchForSuggestionsResult";
const _SFSRL = "SearchForSuggestionsResultList";
const _SFTR = "SearchForTextResult";
const _SFTRL = "SearchForTextResultList";
const _SI = "SensitiveInteger";
const _SL = "StepList";
const _SM = "SubMunicipality";
const _SP = "StartPosition";
const _SPIFP = "SearchPlaceIndexForPosition";
const _SPIFPR = "SearchPlaceIndexForPositionRequest";
const _SPIFPRe = "SearchPlaceIndexForPositionResponse";
const _SPIFPS = "SearchPlaceIndexForPositionSummary";
const _SPIFS = "SearchPlaceIndexForSuggestions";
const _SPIFSR = "SearchPlaceIndexForSuggestionsRequest";
const _SPIFSRe = "SearchPlaceIndexForSuggestionsResponse";
const _SPIFSS = "SearchPlaceIndexForSuggestionsSummary";
const _SPIFT = "SearchPlaceIndexForText";
const _SPIFTR = "SearchPlaceIndexForTextRequest";
const _SPIFTRe = "SearchPlaceIndexForTextResponse";
const _SPIFTS = "SearchPlaceIndexForTextSummary";
const _SQEE = "ServiceQuotaExceededException";
const _SR = "SubRegion";
const _SS = "SensitiveString";
const _ST = "SampleTime";
const _STI = "StartTimeInclusive";
const _SU = "SpeedUnit";
const _Sp = "Speed";
const _St = "Status";
const _Ste = "Steps";
const _Step = "Step";
const _Str = "Street";
const _Sty = "Style";
const _Su = "Summary";
const _T = "Timestamp";
const _TA = "TrackerArn";
const _TAi = "TimingAdvance";
const _TD = "TruckDimensions";
const _TE = "ThrottlingException";
const _TFG = "TrackingFilterGeometry";
const _THM = "TimeHorizonMinutes";
const _TK = "TagKeys";
const _TM = "TravelMode";
const _TMO = "TruckModeOptions";
const _TN = "TrackerName";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TW = "TruckWeight";
const _TZ = "TimeZone";
const _Ta = "Tags";
const _Tac = "Tac";
const _Te = "Text";
const _To = "Total";
const _U = "Updates";
const _UGC = "UpdateGeofenceCollection";
const _UGCR = "UpdateGeofenceCollectionRequest";
const _UGCRp = "UpdateGeofenceCollectionResponse";
const _UK = "UpdateKey";
const _UKR = "UpdateKeyRequest";
const _UKRp = "UpdateKeyResponse";
const _UM = "UpdateMap";
const _UMR = "UpdateMapRequest";
const _UMRp = "UpdateMapResponse";
const _UN = "UnitNumber";
const _UPI = "UpdatePlaceIndex";
const _UPIR = "UpdatePlaceIndexRequest";
const _UPIRp = "UpdatePlaceIndexResponse";
const _UR = "UntagResource";
const _URC = "UpdateRouteCalculator";
const _URCR = "UpdateRouteCalculatorRequest";
const _URCRp = "UpdateRouteCalculatorResponse";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _UT = "UpdateTime";
const _UTR = "UpdateTrackerRequest";
const _UTRp = "UpdateTrackerResponse";
const _UTn = "UnitType";
const _UTp = "UpdateTracker";
const _Un = "Unit";
const _VDP = "VerifyDevicePosition";
const _VDPR = "VerifyDevicePositionRequest";
const _VDPRe = "VerifyDevicePositionResponse";
const _VE = "ValidationException";
const _VEF = "ValidationExceptionField";
const _VEFL = "ValidationExceptionFieldList";
const _W = "Weight";
const _WFAP = "WiFiAccessPoints";
const _WFAPL = "WiFiAccessPointList";
const _WFAPi = "WiFiAccessPoint";
const _WP = "WaypointPositions";
const _WPL = "WaypointPositionList";
const _Wi = "Width";
const _X = "X";
const _Y = "Y";
const _Z = "Z";
const _c = "client";
const _e = "error";
const _en = "endpoint";
const _fD = "forceDelete";
const _fL = "fieldList";
const _h = "http";
const _hE = "httpError";
const _hH = "httpHeader";
const _hQ = "httpQuery";
const _jN = "jsonName";
const _k = "key";
const _l = "language";
const _m = "message";
const _n = "name";
const _r = "reason";
const _s = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.location";
const _tK = "tagKeys";
const n0 = "com.amazonaws.location";
import { TypeRegistry } from "@smithy/core/schema";
import { AccessDeniedException as __AccessDeniedException, ConflictException as __ConflictException, InternalServerException as __InternalServerException, ResourceNotFoundException as __ResourceNotFoundException, ServiceQuotaExceededException as __ServiceQuotaExceededException, ThrottlingException as __ThrottlingException, ValidationException as __ValidationException, } from "../models/index";
import { LocationServiceException as __LocationServiceException } from "../models/LocationServiceException";
export var ApiKey = [0, n0, _AK, 8, 0];
export var Base64EncodedGeobuf = [0, n0, _BEG, 8, 21];
export var CountryCode3 = [0, n0, _CC, 8, 0];
export var CountryCode3OrEmpty = [0, n0, _CCOE, 8, 0];
export var PlaceCategory = [0, n0, _PC, 8, 0];
export var PlaceId = [0, n0, _PI, 8, 0];
export var PlaceSupplementalCategory = [0, n0, _PSC, 8, 0];
export var RefererPattern = [0, n0, _RP, 8, 0];
export var SensitiveBoolean = [0, n0, _SB, 8, 2];
export var SensitiveDouble = [0, n0, _SD, 8, 1];
export var SensitiveInteger = [0, n0, _SI, 8, 1];
export var SensitiveString = [0, n0, _SS, 8, 0];
export var Timestamp = [0, n0, _T, 8, 5];
export var AccessDeniedException = [
    -3,
    n0,
    _ADE,
    {
        [_e]: _c,
        [_hE]: 403,
    },
    [_M],
    [
        [
            0,
            {
                [_jN]: _m,
            },
        ],
    ],
];
TypeRegistry.for(n0).registerError(AccessDeniedException, __AccessDeniedException);
export var AndroidApp = [3, n0, _AA, 0, [_P, _CF], [0, 0]];
export var ApiKeyFilter = [3, n0, _AKF, 0, [_KS], [0]];
export var ApiKeyRestrictions = [
    3,
    n0,
    _AKR,
    0,
    [_AAl, _AR, _ARl, _AAA, _AAAl],
    [64 | 0, 64 | 0, [() => RefererPatternList, 0], () => AndroidAppList, () => AppleAppList],
];
export var AppleApp = [3, n0, _AAp, 0, [_BI], [0]];
export var AssociateTrackerConsumerRequest = [3, n0, _ATCR, 0, [_TN, _CA], [[0, 1], 0]];
export var AssociateTrackerConsumerResponse = [3, n0, _ATCRs, 0, [], []];
export var BatchDeleteDevicePositionHistoryError = [
    3,
    n0,
    _BDDPHE,
    0,
    [_DI, _E],
    [0, () => BatchItemError],
];
export var BatchDeleteDevicePositionHistoryRequest = [
    3,
    n0,
    _BDDPHR,
    0,
    [_TN, _DIe],
    [[0, 1], 64 | 0],
];
export var BatchDeleteDevicePositionHistoryResponse = [
    3,
    n0,
    _BDDPHRa,
    0,
    [_Er],
    [() => BatchDeleteDevicePositionHistoryErrorList],
];
export var BatchDeleteGeofenceError = [3, n0, _BDGE, 0, [_GI, _E], [0, () => BatchItemError]];
export var BatchDeleteGeofenceRequest = [3, n0, _BDGR, 0, [_CN, _GIe], [[0, 1], 64 | 0]];
export var BatchDeleteGeofenceResponse = [
    3,
    n0,
    _BDGRa,
    0,
    [_Er],
    [() => BatchDeleteGeofenceErrorList],
];
export var BatchEvaluateGeofencesError = [
    3,
    n0,
    _BEGE,
    0,
    [_DI, _ST, _E],
    [0, [() => Timestamp, 0], () => BatchItemError],
];
export var BatchEvaluateGeofencesRequest = [
    3,
    n0,
    _BEGR,
    0,
    [_CN, _DPU],
    [
        [0, 1],
        [() => DevicePositionUpdateList, 0],
    ],
];
export var BatchEvaluateGeofencesResponse = [
    3,
    n0,
    _BEGRa,
    0,
    [_Er],
    [[() => BatchEvaluateGeofencesErrorList, 0]],
];
export var BatchGetDevicePositionError = [
    3,
    n0,
    _BGDPE,
    0,
    [_DI, _E],
    [0, () => BatchItemError],
];
export var BatchGetDevicePositionRequest = [3, n0, _BGDPR, 0, [_TN, _DIe], [[0, 1], 64 | 0]];
export var BatchGetDevicePositionResponse = [
    3,
    n0,
    _BGDPRa,
    0,
    [_Er, _DP],
    [() => BatchGetDevicePositionErrorList, [() => DevicePositionList, 0]],
];
export var BatchItemError = [3, n0, _BIE, 0, [_C, _M], [0, 0]];
export var BatchPutGeofenceError = [3, n0, _BPGE, 0, [_GI, _E], [0, () => BatchItemError]];
export var BatchPutGeofenceRequest = [
    3,
    n0,
    _BPGR,
    0,
    [_CN, _En],
    [
        [0, 1],
        [() => BatchPutGeofenceRequestEntryList, 0],
    ],
];
export var BatchPutGeofenceRequestEntry = [
    3,
    n0,
    _BPGRE,
    0,
    [_GI, _G, _GP],
    [0, [() => GeofenceGeometry, 0], [() => PropertyMap, 0]],
];
export var BatchPutGeofenceResponse = [
    3,
    n0,
    _BPGRa,
    0,
    [_S, _Er],
    [[() => BatchPutGeofenceSuccessList, 0], () => BatchPutGeofenceErrorList],
];
export var BatchPutGeofenceSuccess = [
    3,
    n0,
    _BPGS,
    0,
    [_GI, _CT, _UT],
    [0, [() => Timestamp, 0], [() => Timestamp, 0]],
];
export var BatchUpdateDevicePositionError = [
    3,
    n0,
    _BUDPE,
    0,
    [_DI, _ST, _E],
    [0, [() => Timestamp, 0], () => BatchItemError],
];
export var BatchUpdateDevicePositionRequest = [
    3,
    n0,
    _BUDPR,
    0,
    [_TN, _U],
    [
        [0, 1],
        [() => DevicePositionUpdateList, 0],
    ],
];
export var BatchUpdateDevicePositionResponse = [
    3,
    n0,
    _BUDPRa,
    0,
    [_Er],
    [[() => BatchUpdateDevicePositionErrorList, 0]],
];
export var CalculateRouteCarModeOptions = [
    3,
    n0,
    _CRCMO,
    0,
    [_AF, _AT],
    [
        [() => SensitiveBoolean, 0],
        [() => SensitiveBoolean, 0],
    ],
];
export var CalculateRouteMatrixRequest = [
    3,
    n0,
    _CRMR,
    0,
    [_CNa, _DPe, _DPes, _TM, _DT, _DN, _DU, _CMO, _TMO, _K],
    [
        [0, 1],
        [() => PositionList, 0],
        [() => PositionList, 0],
        0,
        [() => Timestamp, 0],
        [() => SensitiveBoolean, 0],
        0,
        [() => CalculateRouteCarModeOptions, 0],
        [() => CalculateRouteTruckModeOptions, 0],
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var CalculateRouteMatrixResponse = [
    3,
    n0,
    _CRMRa,
    0,
    [_RM, _SDP, _SDPn, _Su],
    [[() => RouteMatrix, 0], [() => PositionList, 0], [() => PositionList, 0], () => CalculateRouteMatrixSummary],
];
export var CalculateRouteMatrixSummary = [3, n0, _CRMS, 0, [_DS, _RC, _EC, _DU], [0, 1, 1, 0]];
export var CalculateRouteRequest = [
    3,
    n0,
    _CRR,
    0,
    [_CNa, _DPep, _DPest, _WP, _TM, _DT, _DN, _DU, _ILG, _CMO, _TMO, _ATr, _OF, _K],
    [
        [0, 1],
        [() => Position, 0],
        [() => Position, 0],
        [() => WaypointPositionList, 0],
        0,
        [() => Timestamp, 0],
        [() => SensitiveBoolean, 0],
        0,
        [() => SensitiveBoolean, 0],
        [() => CalculateRouteCarModeOptions, 0],
        [() => CalculateRouteTruckModeOptions, 0],
        [() => Timestamp, 0],
        0,
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var CalculateRouteResponse = [
    3,
    n0,
    _CRRa,
    0,
    [_L, _Su],
    [
        [() => LegList, 0],
        [() => CalculateRouteSummary, 0],
    ],
];
export var CalculateRouteSummary = [
    3,
    n0,
    _CRS,
    0,
    [_RBB, _DS, _D, _DSu, _DU],
    [[() => BoundingBox, 0], 0, [() => SensitiveDouble, 0], [() => SensitiveDouble, 0], 0],
];
export var CalculateRouteTruckModeOptions = [
    3,
    n0,
    _CRTMO,
    0,
    [_AF, _AT, _Di, _W],
    [
        [() => SensitiveBoolean, 0],
        [() => SensitiveBoolean, 0],
        [() => TruckDimensions, 0],
        [() => TruckWeight, 0],
    ],
];
export var CellSignals = [3, n0, _CS, 0, [_LCD], [() => LteCellDetailsList]];
export var Circle = [
    3,
    n0,
    _Ci,
    8,
    [_Ce, _R],
    [
        [() => Position, 0],
        [() => SensitiveDouble, 0],
    ],
];
export var ConflictException = [
    -3,
    n0,
    _CE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_M],
    [
        [
            0,
            {
                [_jN]: _m,
            },
        ],
    ],
];
TypeRegistry.for(n0).registerError(ConflictException, __ConflictException);
export var CreateGeofenceCollectionRequest = [
    3,
    n0,
    _CGCR,
    0,
    [_CN, _PP, _PPDS, _De, _Ta, _KKI],
    [0, 0, 0, 0, 128 | 0, 0],
];
export var CreateGeofenceCollectionResponse = [
    3,
    n0,
    _CGCRr,
    0,
    [_CN, _CAo, _CT],
    [0, 0, [() => Timestamp, 0]],
];
export var CreateKeyRequest = [
    3,
    n0,
    _CKR,
    0,
    [_KN, _Re, _De, _ET, _NE, _Ta],
    [0, [() => ApiKeyRestrictions, 0], 0, [() => Timestamp, 0], 2, 128 | 0],
];
export var CreateKeyResponse = [
    3,
    n0,
    _CKRr,
    0,
    [_K, _KA, _KN, _CT],
    [[() => ApiKey, 0], 0, 0, [() => Timestamp, 0]],
];
export var CreateMapRequest = [
    3,
    n0,
    _CMR,
    0,
    [_MN, _Co, _PP, _De, _Ta],
    [0, [() => MapConfiguration, 0], 0, 0, 128 | 0],
];
export var CreateMapResponse = [3, n0, _CMRr, 0, [_MN, _MA, _CT], [0, 0, [() => Timestamp, 0]]];
export var CreatePlaceIndexRequest = [
    3,
    n0,
    _CPIR,
    0,
    [_IN, _DS, _PP, _De, _DSC, _Ta],
    [0, 0, 0, 0, () => DataSourceConfiguration, 128 | 0],
];
export var CreatePlaceIndexResponse = [
    3,
    n0,
    _CPIRr,
    0,
    [_IN, _IA, _CT],
    [0, 0, [() => Timestamp, 0]],
];
export var CreateRouteCalculatorRequest = [
    3,
    n0,
    _CRCR,
    0,
    [_CNa, _DS, _PP, _De, _Ta],
    [0, 0, 0, 0, 128 | 0],
];
export var CreateRouteCalculatorResponse = [
    3,
    n0,
    _CRCRr,
    0,
    [_CNa, _CAa, _CT],
    [0, 0, [() => Timestamp, 0]],
];
export var CreateTrackerRequest = [
    3,
    n0,
    _CTR,
    0,
    [_TN, _PP, _KKI, _PPDS, _De, _Ta, _PF, _EBE, _KKEGQ],
    [0, 0, 0, 0, 0, 128 | 0, 0, 2, 2],
];
export var CreateTrackerResponse = [
    3,
    n0,
    _CTRr,
    0,
    [_TN, _TA, _CT],
    [0, 0, [() => Timestamp, 0]],
];
export var DataSourceConfiguration = [3, n0, _DSC, 0, [_IU], [0]];
export var DeleteGeofenceCollectionRequest = [3, n0, _DGCR, 0, [_CN], [[0, 1]]];
export var DeleteGeofenceCollectionResponse = [3, n0, _DGCRe, 0, [], []];
export var DeleteKeyRequest = [
    3,
    n0,
    _DKR,
    0,
    [_KN, _FD],
    [
        [0, 1],
        [
            2,
            {
                [_hQ]: _fD,
            },
        ],
    ],
];
export var DeleteKeyResponse = [3, n0, _DKRe, 0, [], []];
export var DeleteMapRequest = [3, n0, _DMR, 0, [_MN], [[0, 1]]];
export var DeleteMapResponse = [3, n0, _DMRe, 0, [], []];
export var DeletePlaceIndexRequest = [3, n0, _DPIR, 0, [_IN], [[0, 1]]];
export var DeletePlaceIndexResponse = [3, n0, _DPIRe, 0, [], []];
export var DeleteRouteCalculatorRequest = [3, n0, _DRCR, 0, [_CNa], [[0, 1]]];
export var DeleteRouteCalculatorResponse = [3, n0, _DRCRe, 0, [], []];
export var DeleteTrackerRequest = [3, n0, _DTR, 0, [_TN], [[0, 1]]];
export var DeleteTrackerResponse = [3, n0, _DTRe, 0, [], []];
export var DescribeGeofenceCollectionRequest = [3, n0, _DGCRes, 0, [_CN], [[0, 1]]];
export var DescribeGeofenceCollectionResponse = [
    3,
    n0,
    _DGCResc,
    0,
    [_CN, _CAo, _De, _PP, _PPDS, _KKI, _Ta, _CT, _UT, _GC],
    [0, 0, 0, 0, 0, 0, 128 | 0, [() => Timestamp, 0], [() => Timestamp, 0], 1],
];
export var DescribeKeyRequest = [3, n0, _DKRes, 0, [_KN], [[0, 1]]];
export var DescribeKeyResponse = [
    3,
    n0,
    _DKResc,
    0,
    [_K, _KA, _KN, _Re, _CT, _ET, _UT, _De, _Ta],
    [
        [() => ApiKey, 0],
        0,
        0,
        [() => ApiKeyRestrictions, 0],
        [() => Timestamp, 0],
        [() => Timestamp, 0],
        [() => Timestamp, 0],
        0,
        128 | 0,
    ],
];
export var DescribeMapRequest = [3, n0, _DMRes, 0, [_MN], [[0, 1]]];
export var DescribeMapResponse = [
    3,
    n0,
    _DMResc,
    0,
    [_MN, _MA, _PP, _DS, _Co, _De, _Ta, _CT, _UT],
    [0, 0, 0, 0, [() => MapConfiguration, 0], 0, 128 | 0, [() => Timestamp, 0], [() => Timestamp, 0]],
];
export var DescribePlaceIndexRequest = [3, n0, _DPIRes, 0, [_IN], [[0, 1]]];
export var DescribePlaceIndexResponse = [
    3,
    n0,
    _DPIResc,
    0,
    [_IN, _IA, _PP, _De, _CT, _UT, _DS, _DSC, _Ta],
    [0, 0, 0, 0, [() => Timestamp, 0], [() => Timestamp, 0], 0, () => DataSourceConfiguration, 128 | 0],
];
export var DescribeRouteCalculatorRequest = [3, n0, _DRCRes, 0, [_CNa], [[0, 1]]];
export var DescribeRouteCalculatorResponse = [
    3,
    n0,
    _DRCResc,
    0,
    [_CNa, _CAa, _PP, _De, _CT, _UT, _DS, _Ta],
    [0, 0, 0, 0, [() => Timestamp, 0], [() => Timestamp, 0], 0, 128 | 0],
];
export var DescribeTrackerRequest = [3, n0, _DTRes, 0, [_TN], [[0, 1]]];
export var DescribeTrackerResponse = [
    3,
    n0,
    _DTResc,
    0,
    [_TN, _TA, _De, _PP, _PPDS, _Ta, _CT, _UT, _KKI, _PF, _EBE, _KKEGQ],
    [0, 0, 0, 0, 0, 128 | 0, [() => Timestamp, 0], [() => Timestamp, 0], 0, 0, 2, 2],
];
export var DevicePosition = [
    3,
    n0,
    _DPev,
    0,
    [_DI, _ST, _RT, _Po, _A, _PPo],
    [
        0,
        [() => Timestamp, 0],
        [() => Timestamp, 0],
        [() => Position, 0],
        [() => PositionalAccuracy, 0],
        [() => PositionPropertyMap, 0],
    ],
];
export var DevicePositionUpdate = [
    3,
    n0,
    _DPUe,
    0,
    [_DI, _ST, _Po, _A, _PPo],
    [0, [() => Timestamp, 0], [() => Position, 0], [() => PositionalAccuracy, 0], [() => PositionPropertyMap, 0]],
];
export var DeviceState = [
    3,
    n0,
    _DSe,
    0,
    [_DI, _ST, _Po, _A, _IAp, _WFAP, _CS],
    [
        0,
        [() => Timestamp, 0],
        [() => Position, 0],
        [() => PositionalAccuracy, 0],
        0,
        () => WiFiAccessPointList,
        () => CellSignals,
    ],
];
export var DisassociateTrackerConsumerRequest = [
    3,
    n0,
    _DTCR,
    0,
    [_TN, _CA],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DisassociateTrackerConsumerResponse = [3, n0, _DTCRi, 0, [], []];
export var ForecastedEvent = [
    3,
    n0,
    _FE,
    0,
    [_EI, _GI, _IDIG, _ND, _ETv, _FBT, _GP],
    [0, 0, 2, 1, 0, [() => Timestamp, 0], [() => PropertyMap, 0]],
];
export var ForecastGeofenceEventsDeviceState = [
    3,
    n0,
    _FGEDS,
    0,
    [_Po, _Sp],
    [[() => Position, 0], 1],
];
export var ForecastGeofenceEventsRequest = [
    3,
    n0,
    _FGER,
    0,
    [_CN, _DSe, _THM, _DU, _SU, _NT, _MR],
    [[0, 1], [() => ForecastGeofenceEventsDeviceState, 0], 1, 0, 0, 0, 1],
];
export var ForecastGeofenceEventsResponse = [
    3,
    n0,
    _FGERo,
    0,
    [_FEo, _NT, _DU, _SU],
    [[() => ForecastedEventsList, 0], 0, 0, 0],
];
export var GeofenceGeometry = [
    3,
    n0,
    _GG,
    0,
    [_Pol, _Ci, _Ge, _MP],
    [
        [() => LinearRings, 0],
        [() => Circle, 0],
        [() => Base64EncodedGeobuf, 0],
        [() => MultiLinearRings, 0],
    ],
];
export var GetDevicePositionHistoryRequest = [
    3,
    n0,
    _GDPHR,
    0,
    [_TN, _DI, _NT, _STI, _ETE, _MR],
    [[0, 1], [0, 1], 0, [() => Timestamp, 0], [() => Timestamp, 0], 1],
];
export var GetDevicePositionHistoryResponse = [
    3,
    n0,
    _GDPHRe,
    0,
    [_DP, _NT],
    [[() => DevicePositionList, 0], 0],
];
export var GetDevicePositionRequest = [
    3,
    n0,
    _GDPR,
    0,
    [_TN, _DI],
    [
        [0, 1],
        [0, 1],
    ],
];
export var GetDevicePositionResponse = [
    3,
    n0,
    _GDPRe,
    0,
    [_DI, _ST, _RT, _Po, _A, _PPo],
    [
        0,
        [() => Timestamp, 0],
        [() => Timestamp, 0],
        [() => Position, 0],
        [() => PositionalAccuracy, 0],
        [() => PositionPropertyMap, 0],
    ],
];
export var GetGeofenceRequest = [
    3,
    n0,
    _GGR,
    0,
    [_CN, _GI],
    [
        [0, 1],
        [0, 1],
    ],
];
export var GetGeofenceResponse = [
    3,
    n0,
    _GGRe,
    0,
    [_GI, _G, _St, _CT, _UT, _GP],
    [0, [() => GeofenceGeometry, 0], 0, [() => Timestamp, 0], [() => Timestamp, 0], [() => PropertyMap, 0]],
];
export var GetMapGlyphsRequest = [
    3,
    n0,
    _GMGR,
    0,
    [_MN, _FS, _FUR, _K],
    [
        [0, 1],
        [0, 1],
        [0, 1],
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var GetMapGlyphsResponse = [
    3,
    n0,
    _GMGRe,
    0,
    [_B, _CTo, _CCa],
    [
        [21, 16],
        [
            0,
            {
                [_hH]: _CT_,
            },
        ],
        [
            0,
            {
                [_hH]: _CC_,
            },
        ],
    ],
];
export var GetMapSpritesRequest = [
    3,
    n0,
    _GMSR,
    0,
    [_MN, _FN, _K],
    [
        [0, 1],
        [0, 1],
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var GetMapSpritesResponse = [
    3,
    n0,
    _GMSRe,
    0,
    [_B, _CTo, _CCa],
    [
        [21, 16],
        [
            0,
            {
                [_hH]: _CT_,
            },
        ],
        [
            0,
            {
                [_hH]: _CC_,
            },
        ],
    ],
];
export var GetMapStyleDescriptorRequest = [
    3,
    n0,
    _GMSDR,
    0,
    [_MN, _K],
    [
        [0, 1],
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var GetMapStyleDescriptorResponse = [
    3,
    n0,
    _GMSDRe,
    0,
    [_B, _CTo, _CCa],
    [
        [21, 16],
        [
            0,
            {
                [_hH]: _CT_,
            },
        ],
        [
            0,
            {
                [_hH]: _CC_,
            },
        ],
    ],
];
export var GetMapTileRequest = [
    3,
    n0,
    _GMTR,
    0,
    [_MN, _Z, _X, _Y, _K],
    [
        [0, 1],
        [() => SensitiveString, 1],
        [() => SensitiveString, 1],
        [() => SensitiveString, 1],
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var GetMapTileResponse = [
    3,
    n0,
    _GMTRe,
    0,
    [_B, _CTo, _CCa],
    [
        [21, 16],
        [
            0,
            {
                [_hH]: _CT_,
            },
        ],
        [
            0,
            {
                [_hH]: _CC_,
            },
        ],
    ],
];
export var GetPlaceRequest = [
    3,
    n0,
    _GPR,
    0,
    [_IN, _PI, _La, _K],
    [
        [0, 1],
        [() => PlaceId, 1],
        [
            0,
            {
                [_hQ]: _l,
            },
        ],
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var GetPlaceResponse = [3, n0, _GPRe, 0, [_Pl], [[() => Place, 0]]];
export var InferredState = [
    3,
    n0,
    _IS,
    0,
    [_Po, _A, _DD, _PD],
    [[() => Position, 0], [() => PositionalAccuracy, 0], 1, 2],
];
export var InternalServerException = [
    -3,
    n0,
    _ISE,
    {
        [_e]: _s,
        [_hE]: 500,
    },
    [_M],
    [
        [
            0,
            {
                [_jN]: _m,
            },
        ],
    ],
];
TypeRegistry.for(n0).registerError(InternalServerException, __InternalServerException);
export var Leg = [
    3,
    n0,
    _Le,
    0,
    [_SP, _EP, _D, _DSu, _G, _Ste],
    [
        [() => Position, 0],
        [() => Position, 0],
        [() => SensitiveDouble, 0],
        [() => SensitiveDouble, 0],
        [() => LegGeometry, 0],
        [() => StepList, 0],
    ],
];
export var LegGeometry = [3, n0, _LG, 0, [_LS], [[() => LineString, 0]]];
export var ListDevicePositionsRequest = [
    3,
    n0,
    _LDPR,
    0,
    [_TN, _MR, _NT, _FG],
    [[0, 1], 1, 0, [() => TrackingFilterGeometry, 0]],
];
export var ListDevicePositionsResponse = [
    3,
    n0,
    _LDPRi,
    0,
    [_En, _NT],
    [[() => ListDevicePositionsResponseEntryList, 0], 0],
];
export var ListDevicePositionsResponseEntry = [
    3,
    n0,
    _LDPRE,
    0,
    [_DI, _ST, _Po, _A, _PPo],
    [0, [() => Timestamp, 0], [() => Position, 0], [() => PositionalAccuracy, 0], [() => PositionPropertyMap, 0]],
];
export var ListGeofenceCollectionsRequest = [3, n0, _LGCR, 0, [_MR, _NT], [1, 0]];
export var ListGeofenceCollectionsResponse = [
    3,
    n0,
    _LGCRi,
    0,
    [_En, _NT],
    [[() => ListGeofenceCollectionsResponseEntryList, 0], 0],
];
export var ListGeofenceCollectionsResponseEntry = [
    3,
    n0,
    _LGCRE,
    0,
    [_CN, _De, _PP, _PPDS, _CT, _UT],
    [0, 0, 0, 0, [() => Timestamp, 0], [() => Timestamp, 0]],
];
export var ListGeofenceResponseEntry = [
    3,
    n0,
    _LGRE,
    0,
    [_GI, _G, _St, _CT, _UT, _GP],
    [0, [() => GeofenceGeometry, 0], 0, [() => Timestamp, 0], [() => Timestamp, 0], [() => PropertyMap, 0]],
];
export var ListGeofencesRequest = [3, n0, _LGR, 0, [_CN, _NT, _MR], [[0, 1], 0, 1]];
export var ListGeofencesResponse = [
    3,
    n0,
    _LGRi,
    0,
    [_En, _NT],
    [[() => ListGeofenceResponseEntryList, 0], 0],
];
export var ListKeysRequest = [3, n0, _LKR, 0, [_MR, _NT, _F], [1, 0, () => ApiKeyFilter]];
export var ListKeysResponse = [
    3,
    n0,
    _LKRi,
    0,
    [_En, _NT],
    [[() => ListKeysResponseEntryList, 0], 0],
];
export var ListKeysResponseEntry = [
    3,
    n0,
    _LKRE,
    0,
    [_KN, _ET, _De, _Re, _CT, _UT],
    [0, [() => Timestamp, 0], 0, [() => ApiKeyRestrictions, 0], [() => Timestamp, 0], [() => Timestamp, 0]],
];
export var ListMapsRequest = [3, n0, _LMR, 0, [_MR, _NT], [1, 0]];
export var ListMapsResponse = [
    3,
    n0,
    _LMRi,
    0,
    [_En, _NT],
    [[() => ListMapsResponseEntryList, 0], 0],
];
export var ListMapsResponseEntry = [
    3,
    n0,
    _LMRE,
    0,
    [_MN, _De, _DS, _PP, _CT, _UT],
    [0, 0, 0, 0, [() => Timestamp, 0], [() => Timestamp, 0]],
];
export var ListPlaceIndexesRequest = [3, n0, _LPIR, 0, [_MR, _NT], [1, 0]];
export var ListPlaceIndexesResponse = [
    3,
    n0,
    _LPIRi,
    0,
    [_En, _NT],
    [[() => ListPlaceIndexesResponseEntryList, 0], 0],
];
export var ListPlaceIndexesResponseEntry = [
    3,
    n0,
    _LPIRE,
    0,
    [_IN, _De, _DS, _PP, _CT, _UT],
    [0, 0, 0, 0, [() => Timestamp, 0], [() => Timestamp, 0]],
];
export var ListRouteCalculatorsRequest = [3, n0, _LRCR, 0, [_MR, _NT], [1, 0]];
export var ListRouteCalculatorsResponse = [
    3,
    n0,
    _LRCRi,
    0,
    [_En, _NT],
    [[() => ListRouteCalculatorsResponseEntryList, 0], 0],
];
export var ListRouteCalculatorsResponseEntry = [
    3,
    n0,
    _LRCRE,
    0,
    [_CNa, _De, _DS, _PP, _CT, _UT],
    [0, 0, 0, 0, [() => Timestamp, 0], [() => Timestamp, 0]],
];
export var ListTagsForResourceRequest = [3, n0, _LTFRR, 0, [_RA], [[0, 1]]];
export var ListTagsForResourceResponse = [3, n0, _LTFRRi, 0, [_Ta], [128 | 0]];
export var ListTrackerConsumersRequest = [3, n0, _LTCR, 0, [_TN, _MR, _NT], [[0, 1], 1, 0]];
export var ListTrackerConsumersResponse = [3, n0, _LTCRi, 0, [_CAon, _NT], [64 | 0, 0]];
export var ListTrackersRequest = [3, n0, _LTR, 0, [_MR, _NT], [1, 0]];
export var ListTrackersResponse = [
    3,
    n0,
    _LTRi,
    0,
    [_En, _NT],
    [[() => ListTrackersResponseEntryList, 0], 0],
];
export var ListTrackersResponseEntry = [
    3,
    n0,
    _LTRE,
    0,
    [_TN, _De, _PP, _PPDS, _CT, _UT],
    [0, 0, 0, 0, [() => Timestamp, 0], [() => Timestamp, 0]],
];
export var LteCellDetails = [
    3,
    n0,
    _LCD,
    0,
    [_CI, _Mc, _Mn, _LI, _NM, _TAi, _NC, _Rs, _Rsr, _Tac],
    [1, 1, 1, () => LteLocalId, () => LteNetworkMeasurementsList, 1, 2, 1, 1, 1],
];
export var LteLocalId = [3, n0, _LLI, 0, [_Ea, _Pc], [1, 1]];
export var LteNetworkMeasurements = [
    3,
    n0,
    _LNM,
    0,
    [_Ea, _CI, _Pc, _Rs, _Rsr],
    [1, 1, 1, 1, 1],
];
export var MapConfiguration = [
    3,
    n0,
    _MC,
    0,
    [_Sty, _PV, _CL],
    [0, [() => CountryCode3, 0], 64 | 0],
];
export var MapConfigurationUpdate = [
    3,
    n0,
    _MCU,
    0,
    [_PV, _CL],
    [[() => CountryCode3OrEmpty, 0], 64 | 0],
];
export var Place = [
    3,
    n0,
    _Pl,
    0,
    [_Lab, _G, _AN, _Str, _N, _Mu, _SR, _Reg, _Cou, _PCo, _I, _TZ, _UTn, _UN, _Ca, _SC, _SM],
    [
        [() => SensitiveString, 0],
        [() => PlaceGeometry, 0],
        [() => SensitiveString, 0],
        [() => SensitiveString, 0],
        [() => SensitiveString, 0],
        [() => SensitiveString, 0],
        [() => SensitiveString, 0],
        [() => SensitiveString, 0],
        [() => SensitiveString, 0],
        [() => SensitiveString, 0],
        [() => SensitiveBoolean, 0],
        [() => TimeZone, 0],
        [() => SensitiveString, 0],
        [() => SensitiveString, 0],
        [() => PlaceCategoryList, 0],
        [() => PlaceSupplementalCategoryList, 0],
        [() => SensitiveString, 0],
    ],
];
export var PlaceGeometry = [3, n0, _PG, 0, [_Poi], [[() => Position, 0]]];
export var PositionalAccuracy = [3, n0, _PA, 0, [_H], [[() => SensitiveDouble, 0]]];
export var PutGeofenceRequest = [
    3,
    n0,
    _PGR,
    0,
    [_CN, _GI, _G, _GP],
    [
        [0, 1],
        [0, 1],
        [() => GeofenceGeometry, 0],
        [() => PropertyMap, 0],
    ],
];
export var PutGeofenceResponse = [
    3,
    n0,
    _PGRu,
    0,
    [_GI, _CT, _UT],
    [0, [() => Timestamp, 0], [() => Timestamp, 0]],
];
export var ResourceNotFoundException = [
    -3,
    n0,
    _RNFE,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_M],
    [
        [
            0,
            {
                [_jN]: _m,
            },
        ],
    ],
];
TypeRegistry.for(n0).registerError(ResourceNotFoundException, __ResourceNotFoundException);
export var RouteMatrixEntry = [
    3,
    n0,
    _RME,
    0,
    [_D, _DSu, _E],
    [[() => SensitiveDouble, 0], [() => SensitiveDouble, 0], () => RouteMatrixEntryError],
];
export var RouteMatrixEntryError = [3, n0, _RMEE, 0, [_C, _M], [0, 0]];
export var SearchForPositionResult = [
    3,
    n0,
    _SFPR,
    0,
    [_Pl, _D, _PI],
    [
        [() => Place, 0],
        [() => SensitiveDouble, 0],
        [() => PlaceId, 0],
    ],
];
export var SearchForSuggestionsResult = [
    3,
    n0,
    _SFSR,
    0,
    [_Te, _PI, _Ca, _SC],
    [
        [() => SensitiveString, 0],
        [() => PlaceId, 0],
        [() => PlaceCategoryList, 0],
        [() => PlaceSupplementalCategoryList, 0],
    ],
];
export var SearchForTextResult = [
    3,
    n0,
    _SFTR,
    0,
    [_Pl, _D, _Rel, _PI],
    [
        [() => Place, 0],
        [() => SensitiveDouble, 0],
        [() => SensitiveDouble, 0],
        [() => PlaceId, 0],
    ],
];
export var SearchPlaceIndexForPositionRequest = [
    3,
    n0,
    _SPIFPR,
    0,
    [_IN, _Po, _MR, _La, _K],
    [
        [0, 1],
        [() => Position, 0],
        1,
        0,
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var SearchPlaceIndexForPositionResponse = [
    3,
    n0,
    _SPIFPRe,
    0,
    [_Su, _Res],
    [
        [() => SearchPlaceIndexForPositionSummary, 0],
        [() => SearchForPositionResultList, 0],
    ],
];
export var SearchPlaceIndexForPositionSummary = [
    3,
    n0,
    _SPIFPS,
    0,
    [_Po, _MR, _DS, _La],
    [[() => Position, 0], 1, 0, 0],
];
export var SearchPlaceIndexForSuggestionsRequest = [
    3,
    n0,
    _SPIFSR,
    0,
    [_IN, _Te, _BP, _FBB, _FC, _MR, _La, _FCi, _K],
    [
        [0, 1],
        [() => SensitiveString, 0],
        [() => Position, 0],
        [() => BoundingBox, 0],
        [() => CountryCodeList, 0],
        1,
        0,
        [() => FilterPlaceCategoryList, 0],
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var SearchPlaceIndexForSuggestionsResponse = [
    3,
    n0,
    _SPIFSRe,
    0,
    [_Su, _Res],
    [
        [() => SearchPlaceIndexForSuggestionsSummary, 0],
        [() => SearchForSuggestionsResultList, 0],
    ],
];
export var SearchPlaceIndexForSuggestionsSummary = [
    3,
    n0,
    _SPIFSS,
    0,
    [_Te, _BP, _FBB, _FC, _MR, _DS, _La, _FCi],
    [
        [() => SensitiveString, 0],
        [() => Position, 0],
        [() => BoundingBox, 0],
        [() => CountryCodeList, 0],
        1,
        0,
        0,
        [() => FilterPlaceCategoryList, 0],
    ],
];
export var SearchPlaceIndexForTextRequest = [
    3,
    n0,
    _SPIFTR,
    0,
    [_IN, _Te, _BP, _FBB, _FC, _MR, _La, _FCi, _K],
    [
        [0, 1],
        [() => SensitiveString, 0],
        [() => Position, 0],
        [() => BoundingBox, 0],
        [() => CountryCodeList, 0],
        1,
        0,
        [() => FilterPlaceCategoryList, 0],
        [
            () => ApiKey,
            {
                [_hQ]: _k,
            },
        ],
    ],
];
export var SearchPlaceIndexForTextResponse = [
    3,
    n0,
    _SPIFTRe,
    0,
    [_Su, _Res],
    [
        [() => SearchPlaceIndexForTextSummary, 0],
        [() => SearchForTextResultList, 0],
    ],
];
export var SearchPlaceIndexForTextSummary = [
    3,
    n0,
    _SPIFTS,
    0,
    [_Te, _BP, _FBB, _FC, _MR, _RBBe, _DS, _La, _FCi],
    [
        [() => SensitiveString, 0],
        [() => Position, 0],
        [() => BoundingBox, 0],
        [() => CountryCodeList, 0],
        1,
        [() => BoundingBox, 0],
        0,
        0,
        [() => FilterPlaceCategoryList, 0],
    ],
];
export var ServiceQuotaExceededException = [
    -3,
    n0,
    _SQEE,
    {
        [_e]: _c,
        [_hE]: 402,
    },
    [_M],
    [
        [
            0,
            {
                [_jN]: _m,
            },
        ],
    ],
];
TypeRegistry.for(n0).registerError(ServiceQuotaExceededException, __ServiceQuotaExceededException);
export var Step = [
    3,
    n0,
    _Step,
    0,
    [_SP, _EP, _D, _DSu, _GO],
    [[() => Position, 0], [() => Position, 0], [() => SensitiveDouble, 0], [() => SensitiveDouble, 0], 1],
];
export var TagResourceRequest = [3, n0, _TRR, 0, [_RA, _Ta], [[0, 1], 128 | 0]];
export var TagResourceResponse = [3, n0, _TRRa, 0, [], []];
export var ThrottlingException = [
    -3,
    n0,
    _TE,
    {
        [_e]: _c,
        [_hE]: 429,
    },
    [_M],
    [
        [
            0,
            {
                [_jN]: _m,
            },
        ],
    ],
];
TypeRegistry.for(n0).registerError(ThrottlingException, __ThrottlingException);
export var TimeZone = [
    3,
    n0,
    _TZ,
    0,
    [_Na, _O],
    [
        [() => SensitiveString, 0],
        [() => SensitiveInteger, 0],
    ],
];
export var TrackingFilterGeometry = [3, n0, _TFG, 0, [_Pol], [[() => LinearRings, 0]]];
export var TruckDimensions = [
    3,
    n0,
    _TD,
    0,
    [_Len, _He, _Wi, _Un],
    [[() => SensitiveDouble, 0], [() => SensitiveDouble, 0], [() => SensitiveDouble, 0], 0],
];
export var TruckWeight = [3, n0, _TW, 0, [_To, _Un], [[() => SensitiveDouble, 0], 0]];
export var UntagResourceRequest = [
    3,
    n0,
    _URR,
    0,
    [_RA, _TK],
    [
        [0, 1],
        [
            64 | 0,
            {
                [_hQ]: _tK,
            },
        ],
    ],
];
export var UntagResourceResponse = [3, n0, _URRn, 0, [], []];
export var UpdateGeofenceCollectionRequest = [
    3,
    n0,
    _UGCR,
    0,
    [_CN, _PP, _PPDS, _De],
    [[0, 1], 0, 0, 0],
];
export var UpdateGeofenceCollectionResponse = [
    3,
    n0,
    _UGCRp,
    0,
    [_CN, _CAo, _UT],
    [0, 0, [() => Timestamp, 0]],
];
export var UpdateKeyRequest = [
    3,
    n0,
    _UKR,
    0,
    [_KN, _De, _ET, _NE, _FU, _Re],
    [[0, 1], 0, [() => Timestamp, 0], 2, 2, [() => ApiKeyRestrictions, 0]],
];
export var UpdateKeyResponse = [3, n0, _UKRp, 0, [_KA, _KN, _UT], [0, 0, [() => Timestamp, 0]]];
export var UpdateMapRequest = [
    3,
    n0,
    _UMR,
    0,
    [_MN, _PP, _De, _CU],
    [[0, 1], 0, 0, [() => MapConfigurationUpdate, 0]],
];
export var UpdateMapResponse = [3, n0, _UMRp, 0, [_MN, _MA, _UT], [0, 0, [() => Timestamp, 0]]];
export var UpdatePlaceIndexRequest = [
    3,
    n0,
    _UPIR,
    0,
    [_IN, _PP, _De, _DSC],
    [[0, 1], 0, 0, () => DataSourceConfiguration],
];
export var UpdatePlaceIndexResponse = [
    3,
    n0,
    _UPIRp,
    0,
    [_IN, _IA, _UT],
    [0, 0, [() => Timestamp, 0]],
];
export var UpdateRouteCalculatorRequest = [3, n0, _URCR, 0, [_CNa, _PP, _De], [[0, 1], 0, 0]];
export var UpdateRouteCalculatorResponse = [
    3,
    n0,
    _URCRp,
    0,
    [_CNa, _CAa, _UT],
    [0, 0, [() => Timestamp, 0]],
];
export var UpdateTrackerRequest = [
    3,
    n0,
    _UTR,
    0,
    [_TN, _PP, _PPDS, _De, _PF, _EBE, _KKEGQ],
    [[0, 1], 0, 0, 0, 0, 2, 2],
];
export var UpdateTrackerResponse = [
    3,
    n0,
    _UTRp,
    0,
    [_TN, _TA, _UT],
    [0, 0, [() => Timestamp, 0]],
];
export var ValidationException = [
    -3,
    n0,
    _VE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_M, _Rea, _FL],
    [
        [
            0,
            {
                [_jN]: _m,
            },
        ],
        [
            0,
            {
                [_jN]: _r,
            },
        ],
        [
            () => ValidationExceptionFieldList,
            {
                [_jN]: _fL,
            },
        ],
    ],
];
TypeRegistry.for(n0).registerError(ValidationException, __ValidationException);
export var ValidationExceptionField = [
    3,
    n0,
    _VEF,
    0,
    [_Na, _M],
    [
        [
            0,
            {
                [_jN]: _n,
            },
        ],
        [
            0,
            {
                [_jN]: _m,
            },
        ],
    ],
];
export var VerifyDevicePositionRequest = [
    3,
    n0,
    _VDPR,
    0,
    [_TN, _DSe, _DU],
    [[0, 1], [() => DeviceState, 0], 0],
];
export var VerifyDevicePositionResponse = [
    3,
    n0,
    _VDPRe,
    0,
    [_IS, _DI, _ST, _RT, _DU],
    [[() => InferredState, 0], 0, [() => Timestamp, 0], [() => Timestamp, 0], 0],
];
export var WiFiAccessPoint = [3, n0, _WFAPi, 0, [_MAa, _Rss], [0, 1]];
export var LocationServiceException = [-3, _sm, "LocationServiceException", 0, [], []];
TypeRegistry.for(_sm).registerError(LocationServiceException, __LocationServiceException);
export var AndroidAppList = [1, n0, _AAL, 0, () => AndroidApp];
export var ApiKeyActionList = 64 | 0;
export var AppleAppList = [1, n0, _AALp, 0, () => AppleApp];
export var ArnList = 64 | 0;
export var BatchDeleteDevicePositionHistoryErrorList = [
    1,
    n0,
    _BDDPHEL,
    0,
    () => BatchDeleteDevicePositionHistoryError,
];
export var BatchDeleteGeofenceErrorList = [1, n0, _BDGEL, 0, () => BatchDeleteGeofenceError];
export var BatchEvaluateGeofencesErrorList = [
    1,
    n0,
    _BEGEL,
    0,
    [() => BatchEvaluateGeofencesError, 0],
];
export var BatchGetDevicePositionErrorList = [1, n0, _BGDPEL, 0, () => BatchGetDevicePositionError];
export var BatchPutGeofenceErrorList = [1, n0, _BPGEL, 0, () => BatchPutGeofenceError];
export var BatchPutGeofenceRequestEntryList = [
    1,
    n0,
    _BPGREL,
    0,
    [() => BatchPutGeofenceRequestEntry, 0],
];
export var BatchPutGeofenceSuccessList = [1, n0, _BPGSL, 0, [() => BatchPutGeofenceSuccess, 0]];
export var BatchUpdateDevicePositionErrorList = [
    1,
    n0,
    _BUDPEL,
    0,
    [() => BatchUpdateDevicePositionError, 0],
];
export var BoundingBox = [1, n0, _BB, 8, 1];
export var CountryCodeList = [1, n0, _CCL, 0, [() => CountryCode3, 0]];
export var CustomLayerList = 64 | 0;
export var DeviceIdsList = 64 | 0;
export var DevicePositionList = [1, n0, _DPL, 0, [() => DevicePosition, 0]];
export var DevicePositionUpdateList = [1, n0, _DPUL, 0, [() => DevicePositionUpdate, 0]];
export var FilterPlaceCategoryList = [1, n0, _FPCL, 0, [() => PlaceCategory, 0]];
export var ForecastedEventsList = [1, n0, _FEL, 0, [() => ForecastedEvent, 0]];
export var GeoArnList = 64 | 0;
export var IdList = 64 | 0;
export var LegList = [1, n0, _LL, 0, [() => Leg, 0]];
export var LinearRing = [1, n0, _LR, 0, [() => Position, 0]];
export var LinearRings = [1, n0, _LRi, 0, [() => LinearRing, 0]];
export var LineString = [1, n0, _LS, 0, [() => Position, 0]];
export var ListDevicePositionsResponseEntryList = [
    1,
    n0,
    _LDPREL,
    0,
    [() => ListDevicePositionsResponseEntry, 0],
];
export var ListGeofenceCollectionsResponseEntryList = [
    1,
    n0,
    _LGCREL,
    0,
    [() => ListGeofenceCollectionsResponseEntry, 0],
];
export var ListGeofenceResponseEntryList = [1, n0, _LGREL, 0, [() => ListGeofenceResponseEntry, 0]];
export var ListKeysResponseEntryList = [1, n0, _LKREL, 0, [() => ListKeysResponseEntry, 0]];
export var ListMapsResponseEntryList = [1, n0, _LMREL, 0, [() => ListMapsResponseEntry, 0]];
export var ListPlaceIndexesResponseEntryList = [
    1,
    n0,
    _LPIREL,
    0,
    [() => ListPlaceIndexesResponseEntry, 0],
];
export var ListRouteCalculatorsResponseEntryList = [
    1,
    n0,
    _LRCREL,
    0,
    [() => ListRouteCalculatorsResponseEntry, 0],
];
export var ListTrackersResponseEntryList = [1, n0, _LTREL, 0, [() => ListTrackersResponseEntry, 0]];
export var LteCellDetailsList = [1, n0, _LCDL, 0, () => LteCellDetails];
export var LteNetworkMeasurementsList = [1, n0, _LNML, 0, () => LteNetworkMeasurements];
export var MultiLinearRings = [1, n0, _MLR, 0, [() => LinearRings, 0]];
export var PlaceCategoryList = [1, n0, _PCL, 0, [() => PlaceCategory, 0]];
export var PlaceSupplementalCategoryList = [1, n0, _PSCL, 0, [() => PlaceSupplementalCategory, 0]];
export var Position = [1, n0, _Po, 8, 1];
export var PositionList = [1, n0, _PL, 0, [() => Position, 0]];
export var RefererPatternList = [1, n0, _RPL, 0, [() => RefererPattern, 0]];
export var RouteMatrix = [1, n0, _RM, 0, [() => RouteMatrixRow, 0]];
export var RouteMatrixRow = [1, n0, _RMR, 0, [() => RouteMatrixEntry, 0]];
export var SearchForPositionResultList = [1, n0, _SFPRL, 0, [() => SearchForPositionResult, 0]];
export var SearchForSuggestionsResultList = [1, n0, _SFSRL, 0, [() => SearchForSuggestionsResult, 0]];
export var SearchForTextResultList = [1, n0, _SFTRL, 0, [() => SearchForTextResult, 0]];
export var StepList = [1, n0, _SL, 0, [() => Step, 0]];
export var TagKeys = 64 | 0;
export var ValidationExceptionFieldList = [1, n0, _VEFL, 0, [() => ValidationExceptionField, 0]];
export var WaypointPositionList = [1, n0, _WPL, 0, [() => Position, 0]];
export var WiFiAccessPointList = [1, n0, _WFAPL, 0, () => WiFiAccessPoint];
export var PositionPropertyMap = [2, n0, _PPM, 8, 0, 0];
export var PropertyMap = [2, n0, _PM, 8, 0, 0];
export var TagMap = 128 | 0;
export var AssociateTrackerConsumer = [
    9,
    n0,
    _ATC,
    {
        [_h]: ["POST", "/tracking/v0/trackers/{TrackerName}/consumers", 200],
        [_en]: ["cp.tracking."],
    },
    () => AssociateTrackerConsumerRequest,
    () => AssociateTrackerConsumerResponse,
];
export var BatchDeleteDevicePositionHistory = [
    9,
    n0,
    _BDDPH,
    {
        [_h]: ["POST", "/tracking/v0/trackers/{TrackerName}/delete-positions", 200],
        [_en]: ["tracking."],
    },
    () => BatchDeleteDevicePositionHistoryRequest,
    () => BatchDeleteDevicePositionHistoryResponse,
];
export var BatchDeleteGeofence = [
    9,
    n0,
    _BDG,
    {
        [_h]: ["POST", "/geofencing/v0/collections/{CollectionName}/delete-geofences", 200],
        [_en]: ["geofencing."],
    },
    () => BatchDeleteGeofenceRequest,
    () => BatchDeleteGeofenceResponse,
];
export var BatchEvaluateGeofences = [
    9,
    n0,
    _BEGa,
    {
        [_h]: ["POST", "/geofencing/v0/collections/{CollectionName}/positions", 200],
        [_en]: ["geofencing."],
    },
    () => BatchEvaluateGeofencesRequest,
    () => BatchEvaluateGeofencesResponse,
];
export var BatchGetDevicePosition = [
    9,
    n0,
    _BGDP,
    {
        [_h]: ["POST", "/tracking/v0/trackers/{TrackerName}/get-positions", 200],
        [_en]: ["tracking."],
    },
    () => BatchGetDevicePositionRequest,
    () => BatchGetDevicePositionResponse,
];
export var BatchPutGeofence = [
    9,
    n0,
    _BPG,
    {
        [_h]: ["POST", "/geofencing/v0/collections/{CollectionName}/put-geofences", 200],
        [_en]: ["geofencing."],
    },
    () => BatchPutGeofenceRequest,
    () => BatchPutGeofenceResponse,
];
export var BatchUpdateDevicePosition = [
    9,
    n0,
    _BUDP,
    {
        [_h]: ["POST", "/tracking/v0/trackers/{TrackerName}/positions", 200],
        [_en]: ["tracking."],
    },
    () => BatchUpdateDevicePositionRequest,
    () => BatchUpdateDevicePositionResponse,
];
export var CalculateRoute = [
    9,
    n0,
    _CR,
    {
        [_h]: ["POST", "/routes/v0/calculators/{CalculatorName}/calculate/route", 200],
        [_en]: ["routes."],
    },
    () => CalculateRouteRequest,
    () => CalculateRouteResponse,
];
export var CalculateRouteMatrix = [
    9,
    n0,
    _CRM,
    {
        [_h]: ["POST", "/routes/v0/calculators/{CalculatorName}/calculate/route-matrix", 200],
        [_en]: ["routes."],
    },
    () => CalculateRouteMatrixRequest,
    () => CalculateRouteMatrixResponse,
];
export var CreateGeofenceCollection = [
    9,
    n0,
    _CGC,
    {
        [_h]: ["POST", "/geofencing/v0/collections", 200],
        [_en]: ["cp.geofencing."],
    },
    () => CreateGeofenceCollectionRequest,
    () => CreateGeofenceCollectionResponse,
];
export var CreateKey = [
    9,
    n0,
    _CK,
    {
        [_h]: ["POST", "/metadata/v0/keys", 200],
        [_en]: ["cp.metadata."],
    },
    () => CreateKeyRequest,
    () => CreateKeyResponse,
];
export var CreateMap = [
    9,
    n0,
    _CM,
    {
        [_h]: ["POST", "/maps/v0/maps", 200],
        [_en]: ["cp.maps."],
    },
    () => CreateMapRequest,
    () => CreateMapResponse,
];
export var CreatePlaceIndex = [
    9,
    n0,
    _CPI,
    {
        [_h]: ["POST", "/places/v0/indexes", 200],
        [_en]: ["cp.places."],
    },
    () => CreatePlaceIndexRequest,
    () => CreatePlaceIndexResponse,
];
export var CreateRouteCalculator = [
    9,
    n0,
    _CRC,
    {
        [_h]: ["POST", "/routes/v0/calculators", 200],
        [_en]: ["cp.routes."],
    },
    () => CreateRouteCalculatorRequest,
    () => CreateRouteCalculatorResponse,
];
export var CreateTracker = [
    9,
    n0,
    _CTr,
    {
        [_h]: ["POST", "/tracking/v0/trackers", 200],
        [_en]: ["cp.tracking."],
    },
    () => CreateTrackerRequest,
    () => CreateTrackerResponse,
];
export var DeleteGeofenceCollection = [
    9,
    n0,
    _DGC,
    {
        [_h]: ["DELETE", "/geofencing/v0/collections/{CollectionName}", 200],
        [_en]: ["cp.geofencing."],
    },
    () => DeleteGeofenceCollectionRequest,
    () => DeleteGeofenceCollectionResponse,
];
export var DeleteKey = [
    9,
    n0,
    _DK,
    {
        [_h]: ["DELETE", "/metadata/v0/keys/{KeyName}", 200],
        [_en]: ["cp.metadata."],
    },
    () => DeleteKeyRequest,
    () => DeleteKeyResponse,
];
export var DeleteMap = [
    9,
    n0,
    _DM,
    {
        [_h]: ["DELETE", "/maps/v0/maps/{MapName}", 200],
        [_en]: ["cp.maps."],
    },
    () => DeleteMapRequest,
    () => DeleteMapResponse,
];
export var DeletePlaceIndex = [
    9,
    n0,
    _DPI,
    {
        [_h]: ["DELETE", "/places/v0/indexes/{IndexName}", 200],
        [_en]: ["cp.places."],
    },
    () => DeletePlaceIndexRequest,
    () => DeletePlaceIndexResponse,
];
export var DeleteRouteCalculator = [
    9,
    n0,
    _DRC,
    {
        [_h]: ["DELETE", "/routes/v0/calculators/{CalculatorName}", 200],
        [_en]: ["cp.routes."],
    },
    () => DeleteRouteCalculatorRequest,
    () => DeleteRouteCalculatorResponse,
];
export var DeleteTracker = [
    9,
    n0,
    _DTe,
    {
        [_h]: ["DELETE", "/tracking/v0/trackers/{TrackerName}", 200],
        [_en]: ["cp.tracking."],
    },
    () => DeleteTrackerRequest,
    () => DeleteTrackerResponse,
];
export var DescribeGeofenceCollection = [
    9,
    n0,
    _DGCe,
    {
        [_h]: ["GET", "/geofencing/v0/collections/{CollectionName}", 200],
        [_en]: ["cp.geofencing."],
    },
    () => DescribeGeofenceCollectionRequest,
    () => DescribeGeofenceCollectionResponse,
];
export var DescribeKey = [
    9,
    n0,
    _DKe,
    {
        [_h]: ["GET", "/metadata/v0/keys/{KeyName}", 200],
        [_en]: ["cp.metadata."],
    },
    () => DescribeKeyRequest,
    () => DescribeKeyResponse,
];
export var DescribeMap = [
    9,
    n0,
    _DMe,
    {
        [_h]: ["GET", "/maps/v0/maps/{MapName}", 200],
        [_en]: ["cp.maps."],
    },
    () => DescribeMapRequest,
    () => DescribeMapResponse,
];
export var DescribePlaceIndex = [
    9,
    n0,
    _DPIe,
    {
        [_h]: ["GET", "/places/v0/indexes/{IndexName}", 200],
        [_en]: ["cp.places."],
    },
    () => DescribePlaceIndexRequest,
    () => DescribePlaceIndexResponse,
];
export var DescribeRouteCalculator = [
    9,
    n0,
    _DRCe,
    {
        [_h]: ["GET", "/routes/v0/calculators/{CalculatorName}", 200],
        [_en]: ["cp.routes."],
    },
    () => DescribeRouteCalculatorRequest,
    () => DescribeRouteCalculatorResponse,
];
export var DescribeTracker = [
    9,
    n0,
    _DTes,
    {
        [_h]: ["GET", "/tracking/v0/trackers/{TrackerName}", 200],
        [_en]: ["cp.tracking."],
    },
    () => DescribeTrackerRequest,
    () => DescribeTrackerResponse,
];
export var DisassociateTrackerConsumer = [
    9,
    n0,
    _DTC,
    {
        [_h]: ["DELETE", "/tracking/v0/trackers/{TrackerName}/consumers/{ConsumerArn}", 200],
        [_en]: ["cp.tracking."],
    },
    () => DisassociateTrackerConsumerRequest,
    () => DisassociateTrackerConsumerResponse,
];
export var ForecastGeofenceEvents = [
    9,
    n0,
    _FGE,
    {
        [_en]: ["geofencing."],
        [_h]: ["POST", "/geofencing/v0/collections/{CollectionName}/forecast-geofence-events", 200],
    },
    () => ForecastGeofenceEventsRequest,
    () => ForecastGeofenceEventsResponse,
];
export var GetDevicePosition = [
    9,
    n0,
    _GDP,
    {
        [_h]: ["GET", "/tracking/v0/trackers/{TrackerName}/devices/{DeviceId}/positions/latest", 200],
        [_en]: ["tracking."],
    },
    () => GetDevicePositionRequest,
    () => GetDevicePositionResponse,
];
export var GetDevicePositionHistory = [
    9,
    n0,
    _GDPH,
    {
        [_en]: ["tracking."],
        [_h]: ["POST", "/tracking/v0/trackers/{TrackerName}/devices/{DeviceId}/list-positions", 200],
    },
    () => GetDevicePositionHistoryRequest,
    () => GetDevicePositionHistoryResponse,
];
export var GetGeofence = [
    9,
    n0,
    _GGe,
    {
        [_h]: ["GET", "/geofencing/v0/collections/{CollectionName}/geofences/{GeofenceId}", 200],
        [_en]: ["geofencing."],
    },
    () => GetGeofenceRequest,
    () => GetGeofenceResponse,
];
export var GetMapGlyphs = [
    9,
    n0,
    _GMG,
    {
        [_h]: ["GET", "/maps/v0/maps/{MapName}/glyphs/{FontStack}/{FontUnicodeRange}", 200],
        [_en]: ["maps."],
    },
    () => GetMapGlyphsRequest,
    () => GetMapGlyphsResponse,
];
export var GetMapSprites = [
    9,
    n0,
    _GMS,
    {
        [_h]: ["GET", "/maps/v0/maps/{MapName}/sprites/{FileName}", 200],
        [_en]: ["maps."],
    },
    () => GetMapSpritesRequest,
    () => GetMapSpritesResponse,
];
export var GetMapStyleDescriptor = [
    9,
    n0,
    _GMSD,
    {
        [_h]: ["GET", "/maps/v0/maps/{MapName}/style-descriptor", 200],
        [_en]: ["maps."],
    },
    () => GetMapStyleDescriptorRequest,
    () => GetMapStyleDescriptorResponse,
];
export var GetMapTile = [
    9,
    n0,
    _GMT,
    {
        [_h]: ["GET", "/maps/v0/maps/{MapName}/tiles/{Z}/{X}/{Y}", 200],
        [_en]: ["maps."],
    },
    () => GetMapTileRequest,
    () => GetMapTileResponse,
];
export var GetPlace = [
    9,
    n0,
    _GPe,
    {
        [_h]: ["GET", "/places/v0/indexes/{IndexName}/places/{PlaceId}", 200],
        [_en]: ["places."],
    },
    () => GetPlaceRequest,
    () => GetPlaceResponse,
];
export var ListDevicePositions = [
    9,
    n0,
    _LDP,
    {
        [_en]: ["tracking."],
        [_h]: ["POST", "/tracking/v0/trackers/{TrackerName}/list-positions", 200],
    },
    () => ListDevicePositionsRequest,
    () => ListDevicePositionsResponse,
];
export var ListGeofenceCollections = [
    9,
    n0,
    _LGC,
    {
        [_en]: ["cp.geofencing."],
        [_h]: ["POST", "/geofencing/v0/list-collections", 200],
    },
    () => ListGeofenceCollectionsRequest,
    () => ListGeofenceCollectionsResponse,
];
export var ListGeofences = [
    9,
    n0,
    _LGi,
    {
        [_en]: ["geofencing."],
        [_h]: ["POST", "/geofencing/v0/collections/{CollectionName}/list-geofences", 200],
    },
    () => ListGeofencesRequest,
    () => ListGeofencesResponse,
];
export var ListKeys = [
    9,
    n0,
    _LK,
    {
        [_en]: ["cp.metadata."],
        [_h]: ["POST", "/metadata/v0/list-keys", 200],
    },
    () => ListKeysRequest,
    () => ListKeysResponse,
];
export var ListMaps = [
    9,
    n0,
    _LM,
    {
        [_h]: ["POST", "/maps/v0/list-maps", 200],
        [_en]: ["cp.maps."],
    },
    () => ListMapsRequest,
    () => ListMapsResponse,
];
export var ListPlaceIndexes = [
    9,
    n0,
    _LPI,
    {
        [_h]: ["POST", "/places/v0/list-indexes", 200],
        [_en]: ["cp.places."],
    },
    () => ListPlaceIndexesRequest,
    () => ListPlaceIndexesResponse,
];
export var ListRouteCalculators = [
    9,
    n0,
    _LRC,
    {
        [_h]: ["POST", "/routes/v0/list-calculators", 200],
        [_en]: ["cp.routes."],
    },
    () => ListRouteCalculatorsRequest,
    () => ListRouteCalculatorsResponse,
];
export var ListTagsForResource = [
    9,
    n0,
    _LTFR,
    {
        [_h]: ["GET", "/tags/{ResourceArn}", 200],
        [_en]: ["cp.metadata."],
    },
    () => ListTagsForResourceRequest,
    () => ListTagsForResourceResponse,
];
export var ListTrackerConsumers = [
    9,
    n0,
    _LTC,
    {
        [_en]: ["cp.tracking."],
        [_h]: ["POST", "/tracking/v0/trackers/{TrackerName}/list-consumers", 200],
    },
    () => ListTrackerConsumersRequest,
    () => ListTrackerConsumersResponse,
];
export var ListTrackers = [
    9,
    n0,
    _LT,
    {
        [_en]: ["cp.tracking."],
        [_h]: ["POST", "/tracking/v0/list-trackers", 200],
    },
    () => ListTrackersRequest,
    () => ListTrackersResponse,
];
export var PutGeofence = [
    9,
    n0,
    _PGu,
    {
        [_h]: ["PUT", "/geofencing/v0/collections/{CollectionName}/geofences/{GeofenceId}", 200],
        [_en]: ["geofencing."],
    },
    () => PutGeofenceRequest,
    () => PutGeofenceResponse,
];
export var SearchPlaceIndexForPosition = [
    9,
    n0,
    _SPIFP,
    {
        [_h]: ["POST", "/places/v0/indexes/{IndexName}/search/position", 200],
        [_en]: ["places."],
    },
    () => SearchPlaceIndexForPositionRequest,
    () => SearchPlaceIndexForPositionResponse,
];
export var SearchPlaceIndexForSuggestions = [
    9,
    n0,
    _SPIFS,
    {
        [_h]: ["POST", "/places/v0/indexes/{IndexName}/search/suggestions", 200],
        [_en]: ["places."],
    },
    () => SearchPlaceIndexForSuggestionsRequest,
    () => SearchPlaceIndexForSuggestionsResponse,
];
export var SearchPlaceIndexForText = [
    9,
    n0,
    _SPIFT,
    {
        [_h]: ["POST", "/places/v0/indexes/{IndexName}/search/text", 200],
        [_en]: ["places."],
    },
    () => SearchPlaceIndexForTextRequest,
    () => SearchPlaceIndexForTextResponse,
];
export var TagResource = [
    9,
    n0,
    _TR,
    {
        [_h]: ["POST", "/tags/{ResourceArn}", 200],
        [_en]: ["cp.metadata."],
    },
    () => TagResourceRequest,
    () => TagResourceResponse,
];
export var UntagResource = [
    9,
    n0,
    _UR,
    {
        [_h]: ["DELETE", "/tags/{ResourceArn}", 200],
        [_en]: ["cp.metadata."],
    },
    () => UntagResourceRequest,
    () => UntagResourceResponse,
];
export var UpdateGeofenceCollection = [
    9,
    n0,
    _UGC,
    {
        [_h]: ["PATCH", "/geofencing/v0/collections/{CollectionName}", 200],
        [_en]: ["cp.geofencing."],
    },
    () => UpdateGeofenceCollectionRequest,
    () => UpdateGeofenceCollectionResponse,
];
export var UpdateKey = [
    9,
    n0,
    _UK,
    {
        [_h]: ["PATCH", "/metadata/v0/keys/{KeyName}", 200],
        [_en]: ["cp.metadata."],
    },
    () => UpdateKeyRequest,
    () => UpdateKeyResponse,
];
export var UpdateMap = [
    9,
    n0,
    _UM,
    {
        [_h]: ["PATCH", "/maps/v0/maps/{MapName}", 200],
        [_en]: ["cp.maps."],
    },
    () => UpdateMapRequest,
    () => UpdateMapResponse,
];
export var UpdatePlaceIndex = [
    9,
    n0,
    _UPI,
    {
        [_h]: ["PATCH", "/places/v0/indexes/{IndexName}", 200],
        [_en]: ["cp.places."],
    },
    () => UpdatePlaceIndexRequest,
    () => UpdatePlaceIndexResponse,
];
export var UpdateRouteCalculator = [
    9,
    n0,
    _URC,
    {
        [_h]: ["PATCH", "/routes/v0/calculators/{CalculatorName}", 200],
        [_en]: ["cp.routes."],
    },
    () => UpdateRouteCalculatorRequest,
    () => UpdateRouteCalculatorResponse,
];
export var UpdateTracker = [
    9,
    n0,
    _UTp,
    {
        [_h]: ["PATCH", "/tracking/v0/trackers/{TrackerName}", 200],
        [_en]: ["cp.tracking."],
    },
    () => UpdateTrackerRequest,
    () => UpdateTrackerResponse,
];
export var VerifyDevicePosition = [
    9,
    n0,
    _VDP,
    {
        [_h]: ["POST", "/tracking/v0/trackers/{TrackerName}/positions/verify", 200],
        [_en]: ["tracking."],
    },
    () => VerifyDevicePositionRequest,
    () => VerifyDevicePositionResponse,
];
